src/WhitelistNFT.sol

```diff
@@ -31,4 +38,24 @@ contract WhitelistNFT is ERC721, Ownable {
         require(owner != address(0), "ZERO_ADDRESS");
         return super.balanceOf(owner);
     }
+
+    /**
+     * @dev Called internally by _mint, _burn, _transfer and _transferFrom functions
+     * we only allow minting and burning of tokens
+     */
+    function _update(
+        address to,
+        uint256 tokenId,
+        address auth
+    ) internal override returns (address) {
+        address owner = _ownerOf(tokenId);
+
+        // if to and owner are non zero addresses, then it is a transfer
+        // we do not allow transfers
+        if (to != address(0) && owner != address(0)) {
```

Comment:
this can simply be:
`require(to != address(0) && owner != address(0, "Soulbound: Transfer not allowed")`

Reply:
I suggest we do the following. If we directly put `require` as the first statement inside the function, that would mean it will check for _mint and _burn functions as well -- we don't want that. We should first detect if it's a transfer call and revert it. wdyt?

```suggestion
       if (to != address(0) && owner != address(0)) {
            revert("Soulbound: Transfer not allowed");
        }
```